R/time-since-quit.R

```diff
@@ -0,0 +1,77 @@
+#' @title Time since quit smoking
+#' 
+#' @description This function creates a derived variable (time_quit_der) that
+#'  calculates the approximate time a former smoker has quit smoking based on
+#'  various CCHS smoking variables. This variable for CCHS respondents in CCHS
+#'  surveys 2003-2014.
+#'  
+#' @param SMK_09A_B number of years since quitting smoking. Variable asked to
+#'  former daily smokers who quit <3 years ago.
+#' 
+#' @param SMKG09C number of years since quitting smoking. Variable asked to
+#'  former daily smokers who quit >=3 years ago.
+#'  
+#' @return value for time since quit smoking in time_quit_der.
+#' 
+#' @examples 
+#' # Using time_quit_fun() to create pack-years values across CCHS cycles
+#' # time_quit_fun() is specified in variable_details.csv along with the CCHS
+#' # variables and cycles included.
+#'
+#' # To transform time_quit_fun across cycles, use rec_with_table() for each
+#' # CCHS cycle and specify time_quit_fun, along with each smoking variable.
+#' # Then by using bind_rows(), you can combine time_quit_fun across cycles
+#' 
+#' library(cchsflow)
+#' 
+#' time_quit2009_2010 <- rec_with_table(
+#'   cchs2009_2010_p, c(
+#'     "SMK_09A_B", "SMKG09C", "time_quit_der"
+#'   )
+#' )
+#'
+#' head(time_quit2009_2010)
+#'
+#' time_quit2011_2012 <- rec_with_table(
+#'   cchs2011_2012_p, c(
+#'     "SMK_09A_B", "SMKG09C", "time_quit_der"
+#'   )
+#' )
+#'
+#' tail(time_quit2011_2012)
+#'
+#' combined_time_quit <- suppressWarnings(bind_rows(time_quit2009_2010,
+#'  time_quit2011_2012))
+#'
+#' head(combined_time_quit)
+#' tail(combined_time_quit)
+#' @export
+
+time_quit_fun <- function(SMK_09A_B, SMKG09C) {
+  SMKG09C_cont <-
+    if_else2(
+      SMKG09C == 1, 4,
+      if_else2(
+        SMKG09C == 2, 8,
+        if_else2(SMKG09C == 3, 12,
+                 if_else2(SMKG09C == "NA(a)", "NA(a)", "NA(b)"
+                          )
+        )
+      )
+    )
+  tsq_ds <-
+    if_else2(
+      SMK_09A_B == 1, 0.5,
+      if_else2(
+        SMK_09A_B == 2, 1.5,
+        if_else2(
+          SMK_09A_B == 3, 2.5,
+          if_else2(SMK_09A_B == 4, SMKG09C_cont,
+                   if_else2(SMKG09C == "NA(a)", "NA(a)", tagged_na("b")
```

Comment:
I'm unclear why you're using tagged_na here and "NA(b)" in line 57.

Reply:
Small mistake on my part. Categorical variables use "NA(a)"/"NA(b)" while continuous variables use tagged_na. The intermediate SMKG09C_cont variable should also be using tagged_na as well.